src/lib/quoter.js

```diff
@@ -52,9 +52,14 @@ class Quoter {
     const hop = this._findBestHopForAmount(request)
     if (!hop) return null
 
-    // If we know a local route to the destinationAccount, use the local route.
+    // note the confusion between `next`, `destination`, and `final`;
+    // `finalAmount` will be renamed to `destinationAmount` later in
+    // the hopToQuote function.
+    if (hop.sourceAmount && hop.finalAmount) {
```

Comment:
To answer your question, if nextAmount is zero, then sourceAmount is lower than the transaction fee for the first hop. When this is the case, we should still just return that quoteResponse, to tell the user about it. It's no use to do a live (remote) quote.
If finalAmount is undefined, then it means we have a (default) route, but don't know what the curve is for it, so this if statement should not be entered, and it should fall through to the remote quoting code path below.